vite.config.ts

```diff
@@ -0,0 +1,43 @@
+import { defineConfig } from "vite";
+import path from "path";
+import babel from "vite-plugin-babel";
+
+export default defineConfig({
+  esbuild: {
+    jsxFactory: "Peact.createElement",
+    jsxFragment: "Peact.Fragment",
+    jsxInject: `import Peact from "@core/peact"`,
+  },
+  plugins: [
+    babel({
+      babelConfig: {
+        babelrc: false,
+        configFile: false,
+        presets: [["@babel/preset-env"], ["@babel/preset-typescript"]],
+        plugins: [
+          [
+            "@babel/plugin-transform-react-jsx",
+            {
+              runtime: "automatic", // 자동으로 pragma를 사용하도록 설정
+              // importSource: "src/core/peact", // Peact의 경로를 지정
+              // pragma: "Peact.createElement",
+              // pragmaFrag: "Preact.Fragment",
+              throwIfNamespace: false, //  JSX 스펙에서는 허용되지만 React에서는 허용되지 않는 JSX인 경우 에러를 throw할건지 말건지 결정하는 옵션
+            },
+          ],
+          ["@babel/plugin-transform-runtime", { corejs: 3 }],
+        ],
+      },
+    }),
```

Comment:
p2: esbuild에서 아래와 같은 옵션으로 작동할 수 있는지 확인해보셔도 좋을 것 같아요 지금은 불필요한 라이브러리가 들어간 느낌이네요!

```suggestion
  esbuild: {
    jsx: 'transform',
    jsxDev: false,
    jsxImportSource: '@/libs/jsx',
    jsxInject: `import { jsx } from '@/libs/jsx/jsx-runtime'`,
    jsxFactory: 'jsx.toVDOM',
  },
```

Reply:
요구사항에 babel 을 통한 환경 설정이 되어 있어 babel 을 사용했습니다만...ㅠ
esbuild 옵션으로 해결가능한 부분은 babel 라이브러리를 사용하지 않는 방향으로 가면 되는 것일까요?!